algoliasearch-common/src/main/java/com/algolia/search/APIClient.java

```diff
@@ -1435,6 +1437,123 @@ public ABTests getABTests(int offset, int limit) throws AlgoliaException {
+        httpClient.requestWithRetry(
+            new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters"),
+                requestOptions,
+                Clusters.class));
+
+    return result.getClusters();
+  }
+
+  public UserIDs listUserIDs() throws AlgoliaException {
+    return this.listUserIDs(0, 20);
+  }
+
+  public UserIDs listUserIDs(Integer page, Integer hitsPerPage) throws AlgoliaException {
+    return this.listUserIDs(page, hitsPerPage, RequestOptions.empty);
+  }
+
+  public UserIDs listUserIDs(
+      @Nonnull Integer page, @Nonnull Integer hitsPerPage, @Nonnull RequestOptions requestOptions)
+      throws AlgoliaException {
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters", "mapping"),
+                requestOptions,
+                UserIDs.class)
+            .setParameters(
+                ImmutableMap.of("page", page.toString(), "hitsPerPage", hitsPerPage.toString())));
+  }
+
+  public Map<String, List<UserID>> getTopUserID() throws AlgoliaException {
+    return this.getTopUserID(RequestOptions.empty);
+  }
+
+  public Map<String, List<UserID>> getTopUserID(RequestOptions requestOptions)
+      throws AlgoliaException {
+    TopUserResult result =
+        httpClient.requestWithRetry(
+            new AlgoliaRequest<>(
+                HttpMethod.GET,
+                AlgoliaRequestKind.SEARCH_API_READ,
+                Arrays.asList("1", "clusters", "mapping", "top"),
+                requestOptions,
+                TopUserResult.class));
+
+    return result.getTopUsers();
+  }
+
+  public AssignUserID assignUserID(@Nonnull String userID, @Nonnull String clusterName)
+      throws AlgoliaException {
+    return this.assignUserID(userID, clusterName, RequestOptions.empty);
+  }
+
+  public AssignUserID assignUserID(
+      @Nonnull String userID, @Nonnull String clusterName, RequestOptions requestOptions)
+      throws AlgoliaException {
+    requestOptions.addExtraHeader("X-Algolia-User-ID", userID);
+
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+                HttpMethod.POST,
+                AlgoliaRequestKind.SEARCH_API_WRITE,
+                Arrays.asList("1", "clusters", "mapping"),
+                requestOptions,
+                AssignUserID.class)
+            .setData(ImmutableMap.of("cluster", clusterName)));
+  }
+
+  public UserID getUserID(@Nonnull String userID) throws AlgoliaException {
+    return this.getUserID(userID, RequestOptions.empty);
+  }
+
+  public UserID getUserID(@Nonnull String userID, RequestOptions requestOptions)
+      throws AlgoliaException {
+    String encodedUserID = "";
+
+    try {
+      encodedUserID = URLEncoder.encode(userID, "UTF-8");
+    } catch (UnsupportedEncodingException ignore) {
```

Comment:
Shouldn't we throw an exception here?

Reply:
I agree with you @Ant-hem, I'll make those throw an `AlgoliaException` wrapping the `UnsupportedEncodingException` rather than ignoring (bad enough to prevent the HTTP call). Let us know if it was on purpose @julienbourdeau.

Reply:
Good catch @Ant-hem